mysql-test/main/mysql_upgrade-6984.test

```diff
@@ -26,3 +26,7 @@ set global event_scheduler=OFF;
 
 let MYSQLD_DATADIR= `select @@datadir`;
 --remove_file $MYSQLD_DATADIR/mysql_upgrade_info
+
+# --skip-grant-tables state may changed during the test. Need to restart the server
```

Comment:
Hi Daniel. There is a MTR internal check that makes sure the system configuration before and after the test were same. Seems like somewhere in this test changed the "skip-grant-table" state from ON to OFF and did not restore it. It was not seen before because "skip-grant-table" was not part of the system variables, but now it is visible.

Without the restart it looks like this:

> MTR's internal check of the test case 'main.mysql_upgrade-6984' failed.
> This means that the test case does not preserve the state that existed
> before the test case was executed.  Most likely the test case did not
> do a proper clean-up. It could also be caused by the previous test run
> by this thread, if the server wasn't restarted.
> This is the diff of the states of the servers before and after the
> test case was executed:
> 
> --- /workplace/tingynia/upstream_10_9/mariadb/mysql-test/var/tmp/check-mysqld_1.result	2022-05-24 16:02:19.017839552 +0000
> +++ /workplace/tingynia/upstream_10_9/mariadb/mysql-test/var/tmp/check-mysqld_1.reject	2022-05-24 16:02:19.441839061 +0000
> @@ -395,7 +395,7 @@
>  SESSION_TRACK_SYSTEM_VARIABLES	autocommit,character_set_client,character_set_connection,character_set_results,time_zone
>  SESSION_TRACK_TRANSACTION_INFO	OFF
>  SKIP_EXTERNAL_LOCKING	ON
> -SKIP_GRANT_TABLES	ON
> +SKIP_GRANT_TABLES	OFF
>  SKIP_NAME_RESOLVE	OFF
>  SKIP_NETWORKING	OFF
>  SKIP_SHOW_DATABASE	OFF

Reply:
ack. I'll look closer.